common/flatpak-run-dbus.c

```diff
@@ -227,12 +227,96 @@ create_proxy_socket (char *template)
   return g_steal_pointer (&proxy_socket);
 }
 
+static char *
+containers1_add_server (FlatpakBwrap *app_bwrap,
+                        const char   *dbus_address,
+                        const char   *app_id,
+                        const char   *instance_id,
+                        GVariant     *metadata)
+{
+  g_autoptr(GDBusConnection) session_bus = NULL;
+  GVariantBuilder named_args;
+  GVariant *body = NULL;
+  g_autoptr(GDBusMessage) msg = NULL;
+  g_autoptr(GDBusMessage) reply = NULL;
+  g_autoptr(GError) local_error = NULL;
+  GVariant *results = NULL;
+  const char *container_path = NULL;
+  const char *container_dus_address = NULL;
+  int sync_fd = -1;
+  g_autoptr(GUnixFDList) fd_list = NULL;
+  int sync_fd_index = -1;
+
+  session_bus = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, NULL);
+  if (session_bus == NULL)
+    return NULL;
+
+  sync_fd = flatpak_bwrap_add_sync_fd (app_bwrap);
```

Comment:
Yes!

A call to `flatpak_bwrap_add_sync_fd` returns the same fd when called multiple times. This is okay because we use it everywhere to get notified when the app is exiting.